src/Hosting.CommandLine/Internal/CommandLineService{T}.cs

```diff
@@ -26,8 +26,9 @@ internal class CommandLineService<T> : IDisposable, ICommandLineService where T
         /// <param name="logger">A logger</param>
         /// <param name="state">The command line state</param>
         /// <param name="serviceProvider">The DI service provider</param>
+        /// <param name="configure">The delegate to configure the app</param>
         public CommandLineService(ILogger<CommandLineService<T>> logger, CommandLineState state,
-            IServiceProvider serviceProvider)
+            IServiceProvider serviceProvider, Action<CommandLineApplication> configure)
```

Comment:
```suggestion
            IServiceProvider serviceProvider, Action<CommandLineApplication<T>> configure)
```

Reply:
The problem is the `_application` is `CommandLineApplication` but not `CommandLineApplication<T>`

Reply:
Is there a reason we cannot update _application to the generic type?

Reply:
It looks very strange to me.
`_application = new CommandLineApplication<T>(state.Console, state.WorkingDirectory)`
I thought the _application is a generic one but Visual Studio tells me it is not.

Reply:
Never mind. I forgot the variable declaration.